pylint/interfaces.py

```diff
@@ -39,6 +40,7 @@
 UNDEFINED = Confidence("UNDEFINED", "Warning without any associated confidence level.")
 
 CONFIDENCE_LEVELS = [HIGH, CONTROL_FLOW, INFERENCE, INFERENCE_FAILURE, UNDEFINED]
+CONFIDENCE_LEVEL_NAMES = [i.name for i in CONFIDENCE_LEVELS]
```

Comment:
We're really stuck with the legacy API here an enum for Confidence would be so much simpler.  Ho well...